pkg/tfvars/baremetal/baremetal.go

```diff
@@ -85,7 +86,15 @@ func TFVars(libvirtURI, bootstrapProvisioningIP, bootstrapOSImage, externalBridg
 		// Instance Info
 		// The rhcos-downloader container downloads the image, compresses it to speed up deployments
 		// and then makes it available on bootstrapProvisioningIP via http
-		imageFilename := path.Base(image)
+		// The image is now formatted with a query string containing the sha256sum, we strip that here
+		// and it will be consumed for validation in https://github.com/openshift/ironic-rhcos-downloader
+		imageURL, err := url.Parse(image)
+		if err != nil {
+			return nil, err
+		}
+		imageURL.RawQuery = ""
+		imageURL.Fragment = ""
+		imageFilename := path.Base(imageURL.String())
```

Comment:
This doesn't appear to strip out parameters
```
    var osImage = "http://asfafs/dfsdf.qcow2?asd=asda";
    imageURL, _ := url.Parse(osImage)
    fmt.Println(">> "+path.Base(imageURL.String()));
ouputs
>> dfsdf.qcow2?asd=asda

```

Reply:
> This doesn't appear to strip out parameters
> 
> ```
>     var osImage = "http://asfafs/dfsdf.qcow2?asd=asda";
>     imageURL, _ := url.Parse(osImage)
>     fmt.Println(">> "+path.Base(imageURL.String()));
> ouputs
> >> dfsdf.qcow2?asd=asda
> ```

scratch that, I see what your doing now, still trying to figure out why its not working for me

Reply:
> scratch that, I see what your doing now, still trying to figure out why its not working for me

I based it on some testing in https://play.golang.org/p/IeCJ6Xzu70e which you can try, we could also split on the `?` I guess but this approach seemed safer?

Reply:
Duh, Lesson learned, this is how I applied your patch 

```
[notstack@host11 installer]$ git reset --hard origin/master                                                                                                                                                        
HEAD is now at 77d20c8 Merge pull request #2620 from iamemilio/known-issues

[notstack@host11 installer]$ git fetch https://github.com/hardys/installer/ image_fix
From https://github.com/hardys/installer
 * branch            image_fix  -> FETCH_HEAD

[notstack@host11 installer]$ git cherry-pick FETCH_HEAD
[master 7ddfed8] Strip sha256sum query string from baremetal machine providerSpec
 Author: Steven Hardy <shardy@redhat.com>
 2 files changed, 24 insertions(+), 7 deletions(-)
```

but your branch had two patches, cherry-pick only gets one .....

Reply:
Ah yeah, I normally fetch PRs like `git fetch origin pull/123/head:pr_123` to a branch, for exactly that reason :)